url.bs

```diff
@@ -2476,39 +2476,84 @@ background information. [[!HTML]]
+ permits a single scheme (such as a browser feature that omits <code>https://</code> because it is
+ only enabled for secure origins). Otherwise, the scheme may be replaced or supplemented with a
+ human-readable string (e.g., "Not secure"), a security indicator icon, or both.
+
+ <li><p>Browsers may omit a leading <code>www</code> domain label to simplify the host, except when
+ it is part of the <a for=host>registrable domain</a>.
 
- <li><p>Other parts of the <a for=/>URL</a> should have their sequences of
- <a>percent-encoded bytes</a> replaced with code points resulting from
- <a>percent decoding</a> those sequences converted to bytes, unless that renders those
- sequences invisible.
+ <li><p>As described in <a>URL serializer</a>, browsers should not serialize null ports.
 </ul>
 
-<p>For the purposes of bidirectional text it should be rendered as if it were in a
-left-to-right embedding. [[!BIDI]]
+<h4 id=url-rendering-elision>Elision</h4>
 
-<p class="note no-backref">Unfortunately, as rendered <a for=/>URLs</a> are strings and can appear
-anywhere, a specific bidirectional algorithm for rendered <a for=/>URLs</a> would not see wide
-adoption. Bidirectional text interacts with the parts of a <a for=/>URL</a> in ways that can cause
-the rendering to be different from the model. Users of bidirectional languages are thus cautioned
-that this is to be expected, particularly in plain text environments.
+<p>In a space-constrained display, URLs should be elided carefully to avoid misleading the user when
+making a security decision:
 
-<p>Due to the confusion that can arise between a <a for=/>URL</a>'s <a for=url>host</a>
-and <a for=url>path</a> with bidirectional text, browsers are encouraged to only render a
-<a for=/>URL</a>'s <a for=url>host</a> in places where it is important for users to
-distinguish between the two. E.g., users are expected to make trust decisions based on a
-<a for=/>URL</a>'s <a for=url>host</a> rendered in the address bar.
+<ul>
+ <li><p>Browsers should ensure that at least the <a for=host>registrable domain</a> can be shown
+ when the URL is rendered (to avoid showing, e.g., <code>...examplecorp.com</code> when loading
+ <code>https://not-really-examplecorp.com/</code>).
+
+ <li><p>When the full <a for=url>host</a> cannot be rendered, browsers should elide domain labels
+ starting from the lowest-level domain label. For example, <code>examplecorp.com.evil.com</code>
+ should be elided as <code>...com.evil.com</code>, not <code>examplecorp.com...</code>. (Note that
+ bidirectional text means that the lowest-level label may not appear at the left.)
+</ul>
 
+<h4 id=url-rendering-i18n>Internationalization and special characters</h4>
 
+<p>International domain names (IDNs), special characters, and bidirectional text should be handled
+with care to prevent spoofing:
+
+<ul>
+ <li><p>Browser should render a <a for=/>URL</a>'s <a for=url>host</a> using
```

Comment:
Should be “Browsers” (plural), for consistency

Reply:
Done